File: UltraTabbedMdiManager_MoveFocus_CS/UltraTabbedMdiManager_MoveFocus_CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Security.Cryptography;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace UltraTabbedMdiManager_MoveFocus_CS
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            //MDI 子フォームのコンテナーとするプロパティ設定
            this.IsMdiContainer = true;
            //フォームでキーイベントを捕捉するためのプロパティ設定
            this.KeyPreview = true;

            for (int i = 1; i <= 5; i++)
            {
                var myForm = new SampleChildForm();
                myForm.MdiParent = this;
                myForm.Text = "Child Form" + i.ToString();
                myForm.Show();
            }
        }

        private void Form1_KeyDown(object sender, KeyEventArgs e)
        {
            var tabs = this.ultraTabbedMdiManager1.TabGroups[0].Tabs;
            var activeTab = this.ultraTabbedMdiManager1.ActiveTab;

            //Home 押下時に、先頭タブをアクティブ化する。
            if (e.KeyData == Keys.Home)
            {
                tabs[0].Activate();
            }

            //End 押下時に、最終タブをアクティブ化する。
            if (e.KeyData == Keys.End)
            {
                tabs[tabs.Count - 1].Activate();
            }

            //PageUp 押下時に、アクティブタブを次に移動する。
            if (e.KeyData == Keys.PageUp)
            {
                //最終タブがアクティブタブの場合、先頭タブをアクティブ化する。
                if (activeTab.Index == tabs.Count - 1)
                {
                    tabs[0].Activate();
                }
                else
                {
                    tabs[activeTab.Index + 1].Activate();
                }
            }

            //PageDown 押下時に、アクティブタブを前に移動する。
            if (e.KeyData == Keys.PageDown)
            {
                //先頭タブがアクティブタブの場合、最終タブをアクティブ化する。
                if (activeTab.Index == 0)
                {
                    tabs[tabs.Count - 1].Activate();
                }
                else
                {
                    tabs[activeTab.Index - 1].Activate();
                }
            }
        }
    }
}

File: UltraTabbedMdiManager_MoveFocus_VB/UltraTabbedMdiManager_MoveFocus_VB/Form1.vb
Imports System.Drawing.Imaging
Imports System.Windows.Forms.VisualStyles.VisualStyleElement

Public Class Form1
	Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load

		'MDI 子フォームのコンテナーとするプロパティ設定
		IsMdiContainer = True
		'フォームでキーイベントを捕捉するためのプロパティ設定
		KeyPreview = True

		Dim i As Integer
		For i = 1 To 5
			Dim myForm As New SampleChildForm()
			myForm.MdiParent = Me
			myForm.Text = "Child Form" & i.ToString()
			myForm.Show()
		Next i
	End Sub

	Private Sub Form1_KeyDown(sender As Object, e As KeyEventArgs) Handles MyBase.KeyDown
		Dim tabs = Me.UltraTabbedMdiManager1.TabGroups(0).Tabs
		Dim activeTab = Me.UltraTabbedMdiManager1.ActiveTab

		'Home 押下時に、先頭タブをアクティブ化する。
		If e.KeyData = Keys.Home Then
			tabs(0).Activate()
		End If

		'End 押下時に、最終タブをアクティブ化する。
		If e.KeyData = Keys.End Then
			tabs(tabs.Count - 1).Activate()
		End If

		'PageUp 押下時に、アクティブタブを次に移動する。
		If e.KeyData = Keys.PageUp Then
			'最終タブがアクティブタブの場合、先頭タブをアクティブ化する。
			If activeTab.Index = tabs.Count - 1 Then
				tabs(0).Activate()
			Else
				tabs(activeTab.Index + 1).Activate()
			End If
		End If

		'PageDown 押下時に、アクティブタブを前に移動する。
		If e.KeyData = Keys.PageDown Then
			'先頭タブがアクティブタブの場合、最終タブをアクティブ化する。
			If activeTab.Index = 0 Then
				tabs(tabs.Count - 1).Activate()
			Else
				tabs(activeTab.Index - 1).Activate()
			End If
		End If
	End Sub
End Class
